### PR TITLE
Update URL-related interfaces and their tests up to spec

### PR DESCRIPTION
**Pull request by nox**
The URL spec recently changed and the variour "mixins" interfaces are gone,
this commit updates our code and WPT accordingly.

The new expected failures related to HTMLAnchorElement and HTMLAreaElement's
attributes are due to their moving to the HTMLHyperLinkElementUtils interface,
which is not anymore in a separate `<script class=untested>` element.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8008)

### CODE REVIEW

File: components/script/dom/urlhelper.rs [hunk 1]
```diff
@@ -96,24 +84,21 @@ impl UrlHelper {
...
     }
 
-    // https://url.spec.whatwg.org/#dom-urlutils-protocol
     pub fn SetProtocol(url: &mut Url, value: USVString) {
         let mut wrapper = UrlUtilsWrapper { url: url, parser: &UrlParser::new() };
         let _ = wrapper.set_scheme(&value.0);
     }
 
-    // https://html.spec.whatwg.org/multipage/#same-origin
+    // g
```
Comment (dzbarsky) [inline components/script/dom/urlhelper.rs]: ?

**Comment by nox** [comment]
@dzbarsky Fat-fingered this, good catch.

**Comment by jdm** [comment]
@bors-servo: r+
Please note the requests to file followup issues!


---

Reviewed 19 of 19 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[components/script/dom/location.rs, line 73 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8008#-K10lA20czxVsOIm73wJ)** ([raw file](https://github.com/servo/servo/blob/5bc2e27bfbbe09ce7e4d2e49f1556fe801ca73a6/components/script/dom/location.rs#L73)):</sup>
According to the spec we should be invoking Location-object-setter in all of these setter methods. Presumably that should have been the case before, so I'm ok with filing it as a followup.


---

<sup>**[components/script/dom/location.rs, line 102 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8008#-K10idWxjni9b6Q_mNJX)** ([raw file](https://github.com/servo/servo/blob/5bc2e27bfbbe09ce7e4d2e49f1556fe801ca73a6/components/script/dom/location.rs#L102)):</sup>
Please file a followup issue to make this and SetProtocol throw per the spec.


---

<sup>**[components/script/dom/webidls/HTMLAreaElement.webidl, line 16 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8008#-K10iHaykuZE7DjpyNrR)** ([raw file](https://github.com/servo/servo/blob/5bc2e27bfbbe09ce7e4d2e49f1556fe801ca73a6/components/script/dom/webidls/HTMLAreaElement.webidl#L16)):</sup>
What's the purpose of this comment?


---

<sup>**[components/script/dom/webidls/HTMLHyperlinkElementUtils.webidl, line 8 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8008#-K10iLbFsnkBDo6WMIAj)** ([raw file](https://github.com/servo/servo/blob/5bc2e27bfbbe09ce7e4d2e49f1556fe801ca73a6/components/script/dom/webidls/HTMLHyperlinkElementUtils.webidl#L8)):</sup>
Why not at least uncomment the interface and annotations?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8008)

**Comment by jdm** [comment]
Sigh. #7625 strikes again.

**Comment by nox** [comment]
@bors-servo retry

Seems like it's ok again?

**Comment by nox** [comment]
Never mind, misread the bot in the IRC channel…

**Pull request merged**